### PR TITLE
Adds Mind Slave Implants!

### PR DESCRIPTION
**Pull request by fluffe9911**
🆑 Fluffe9911
add: Both greater and regular mind slave implants
/🆑

Theres two types of mindslave implants the greater one which lasts the whole round for 10 tc and the lesser which is 6 tc but only lasts around 20 minutes also while this is mostly a hippie port apparently old old yogs did actually have mindslave implants which is the main reason why im porting them now so yeah

### CODE REVIEW

File: yogstation/code/game/objects/items/implants/implant_mindslave.dm [hunk 1]
```diff
@@ -0,0 +1,97 @@
+/obj/item/implant/mindslave
+	name = "mindslave implant"
+	desc = "Now YOU too can have your very own mindslave! Pop this implant into anybody and they'll obey any command you give for around 15 to 20 minutes."
+	activated = 0
```
Comment (Amelia0010) [inline yogstation/code/game/objects/items/implants/implant_mindslave.dm]: FALSE

File: yogstation/code/game/objects/items/implants/implant_mindslave.dm [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+				<b>Function:</b> Allows user to command anyone implanted to do whatever they want.<BR>
+				<b>Special Features:</b> Person with implant MUST obey any order you give. <BR>
+				<b>Integrity:</b> Implant will last around 15 and 20 minutes."}
+	return dat
+
+/obj/item/implant/mindslave/implant(mob/target,mob/user,silent=0)
+	if(target.mind)
+		if(target == user)
+			to_chat(target,"<span class='notice'>You can't implant yourself!</span>")
+			return 0
```
Comment (Amelia0010) [inline yogstation/code/game/objects/items/implants/implant_mindslave.dm]: FALSE

File: yogstation/code/game/objects/items/implants/implant_mindslave.dm [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+		if(target.mind.current.has_trait(TRAIT_MINDSHIELD))
+			to_chat(target,"<span class='danger'>Your loyalty implant rejects [user]'s mindslave!</span>")
+			to_chat(user,"<span class='danger'>[target] somehow rejects the mindslave implant!</span>")
+			return 0
```
Comment (Amelia0010) [inline yogstation/code/game/objects/items/implants/implant_mindslave.dm]: FALSE

File: yogstation/code/game/objects/items/implants/implant_mindslave.dm [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+				<HR>
+				<b>Implant Details:</b><BR>
+				<b>Function:</b> Allows user to command anyone implanted to do whatever they want.<BR>
+				<b>Special Features:</b> Person with implant MUST obey any order you give. <BR>
+				<b>Integrity:</b> Implant will last around 15 and 20 minutes."}
+	return dat
+
+/obj/item/implant/mindslave/implant(mob/target,mob/user,silent=0)
+	if(target.mind)
+		if(target == user)
```
Comment (Amelia0010) [inline yogstation/code/game/objects/items/implants/implant_mindslave.dm:24]: make it not work on changelings, like prebase https://github.com/yogstation13/yogstation/pull/1070

File: yogstation/code/game/objects/items/implants/implant_mindslave.dm [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+			protect_objective = new /datum/objective/protect
+			protect_objective.owner = target.mind
+			protect_objective.target = user.mind
+			protect_objective.explanation_text = "Protect [user], your mindslave master. Obey any command given by them."
+			var/datum/antagonist/mindslave/M = target.mind.add_antag_datum(/datum/antagonist/mindslave)
+			M.master_name = user.real_name
+			M.master_ckey = user.ckey
+			M.objectives |= protect_objective
+			message_admins("[user]/([user.ckey]) made a mindslave out of [target]/([target.ckey]).")
+			return 1
```
Comment (Amelia0010) [inline yogstation/code/game/objects/items/implants/implant_mindslave.dm]: TRUE

File: yogstation/code/game/objects/items/implants/implant_mindslave.dm [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+	to_chat(user,"<span class='notice'>[target] has no mind!</span>")
+	return 0
```
Comment (Amelia0010) [inline yogstation/code/game/objects/items/implants/implant_mindslave.dm]: FALSE

File: yogstation/code/game/objects/items/implants/implant_mindslave.dm [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+				<b>Integrity:</b> Implant will last forever."}
+	return dat
+
+/obj/item/implant/mindslave/proc/remove_mindslave()
+	if(imp_in)
+		if(imp_in.mind.special_role == "Mindslave")
+			imp_in.mind.special_role = ""
+			SSticker.mode.traitors -= imp_in.mind
+		to_chat(imp_in,"<span class='userdanger'>You feel your free will come back to you! You no longer have to obey your master!</span>")
+		to_chat(imp_in,"<span class='userdanger'>If you were not an antagonist BEFORE being mindslave, then you no longer are one.</span>")
```
Comment (Amelia0010) [inline yogstation/code/game/objects/items/implants/implant_mindslave.dm:90]: Make this check whether or not they are actually an antagonist and only tell them they aren't an antag anymore if they're aren't an antag anymore

File: yogstation/code/game/objects/items/storage/uplink_kits.dm [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+/obj/item/storage/box/syndie_kit/imp_mindslave/PopulateContents()
+	new /obj/item/implanter/mindslave(src)
+
+/obj/item/storage/box/syndie_kit/imp_gmindslave
+	name = "Greater Mindslave Implant (with injector)"
+
+/obj/item/storage/box/syndie_kit/imp_gmindslave/PopulateContents()
+	new /obj/item/implanter/mindslave/greater(src)
+
+/obj/item/storage/box/syndie_kit/cluwnification
```
Comment (Amelia0010) [inline yogstation/code/game/objects/items/storage/uplink_kits.dm]: Atomize your PRs, this has nothing to do with the rest of the PR >:(

File: yogstation/code/modules/uplink/uplink_item.dm [hunk 9]
```diff
@@ -69,11 +69,29 @@
...
+	name = "Mindslave Implant"
+	desc = "An implant injected into another body, forcing the victim to obey any command by the user for around 15 to 20 mintues."
+	exclude_modes = list(/datum/game_mode/nuclear)
+	item = /obj/item/storage/box/syndie_kit/imp_mindslave
+	cost = 6
+	surplus = 20
+
+/datum/uplink_item/implants/greatermindslave
+	name = "Greater Mindslave Implant"
+	desc = "An implant injected into another body, forcing the victim to obey any command by the user, it does not expire like a regular mindslave implant."
```
Comment (Amelia0010) [inline yogstation/code/modules/uplink/uplink_item.dm]:
```suggestion
	desc = "An implant injected into another body, forcing the victim to obey any command by the user. It does not expire, unlike a regular mindslave implant."
```

**Review by Amelia0010 — CHANGES_REQUESTED** [review]

File: yogstation/code/game/objects/items/implants/implant_mindslave.dm (hunk 7 above)
Comment (fluffe9911) [reply yogstation/code/game/objects/items/implants/implant_mindslave.dm:90]: Do I have to that seems like extra effort for very little pay off

**Comment by Moja10** [comment]
Make sure there's a special antag hud icon because before the rebase, mindslaves didn't have an icon so it looked like nonantags doing antag things

**Comment by fluffe9911** [comment]
Nich said he planned on doing his own version im just leaving this up until he does it lol

**Comment by Amelia0010** [comment]
oh yeah I forgot hang on

**Pull request closed without merging**